### DISCUSSION TITLE
Use of metapackages?

### DISCUSSION THREAD

**Discussion by StefanCaldararu** [uwsbel/miniav#102] [General]
Discussion for if people will use the metapackgages... I'm concerned that may be too complicated for people to be interested in using this system. Will post more about this later.

**Comment by StefanCaldararu** [comment]
Ok @AaronYoung5 I thought about this more and have some hot takes. I see a few "realistic" paths forward as what could happen for the future:
- Either we don't have a consistent package management system or people don't like the system that is implemented / it's too complicated. Long story short, the different vehicles diverge in terms of shared packages, which is what we want to avoid. Big issues are people aren't bought into the package system, and it is too complicated for people to want to learn.
- First solution to this issue is that everyone sacrifices some time to learning the development workflow. So far, this has been defined by @AaronYoung5, but for this system to work well it would be important that people participate and contribute (ideas wise), so that we actually make changes to the system that people like. Hard part about this is it requires everyone to sacrifice a sizeable amount of time to just learning the structure, rather than doing the research they are actually interested in (this is more or less my interpretation of some of the points @Huzaifg has made).
- I see this as a potential solution, but I don't think @AaronYoung5 or @Huzaifg are going to like it much: what is done on my team at NASA is essentially that the Flight Software Team implements whatever framework each subsystem needs in they style that they like. This basically means @AaronYoung5 (or myself, or someone else who is interested in learning about the metapackages and full A-stack structure) whenever a vehicle wants a new node, or we want to implement a new a-stack structure, someone who knows the structure well talks to the team that wants this done and helps them implement the structure. This puts the majority of the burden in terms of framework on a few people. If they like this work and are interested in it, this is mostly fine... The major issues with this are that if something goes wrong people won't understand how to fix it themselves, and this role needs to get "recycled" (what happens when people leave from this role?).
- Other solution, which is less thought out, is what I have proposed about each vehicle having their own branch they do development in. I'm not sure I like this because this makes it very easy for people lean more towards option 1 - which is what we are trying to avoid. 

@AaronYoung5 thoughts?

**Reply by AaronYoung5** [reply]
I think it should be a mix of people learning as they go and me/knowledgeable people on the system are there to help + having good documentation. I think we should just keep on the same course we're moving on now, if it becomes an issue for people as they get more into the system, we can revist.

**Reply by Huzaifg** [reply]
@StefanCaldararu 
I don't think we can find someone willing to do the conversion of A-stack into standard structure. I agree with Aaron, as people work on this more, they will start understanding the structure more and if people get stuck, we can always ask you/Aaron. I also agree that this means that there is a higher chance of branches diverging beyond repair - maybe to prevent this we should have some sort of monthly checks to see how far off each branch is? 

Also, I am reading parts of the documentation every day, its pretty good :)

Edit: Maybe one of the three meetings should have a chunk of time allocated to ART/ATK issues where people can ask questions and get clarifications. I think most of the confusion sometimes is just lack of communication?

**Reply by StefanCaldararu** [reply]
Yes I agree with all of that, makes a lot of sense to me!
